### ISSUE TITLE
Rewrite fully qualified package name used as argument parameter inside annotation when using Change Package Name rewrite rule

### ISSUE THREAD

**Issue by apupier** [openrewrite/rewrite#3200]
I think that Change package name rule https://docs.openrewrite.org/recipes/java/changepackage#source is ignoring the packages which are inside annotations.

I see no reference of annotation in the implementation class: https://github.com/openrewrite/rewrite/blob/v7.40.5/rewrite-java/src/main/java/org/openrewrite/java/ChangePackage.java

I think it is what it is used by the Quarkus OpenRewrite recipe. I reported intially in Quarkus repository: https://github.com/quarkusio/quarkus/issues/33157

**Comment by timtebeek** [comment]
Could you provide a recipe configuration, along with a code sample that illustrates the issue? I'm having a bit of difficulty visualizing the issue, and a clear example could help reproduce, and rule out configuration issues.

We have [unit tests that look to cover what you're reporting here](https://github.com/openrewrite/rewrite/blob/b9a0475eb72a5cdde5003a64f57bae0a9bf08cb4/rewrite-java-test/src/test/java/org/openrewrite/java/ChangePackageTest.java#L514-L553), so a similar test that fails would be a ideal.

**Comment by apupier** [comment]
I'm not familiar with OpenRewrite neither Gradle and it was only as a side tasks that I noticed that so it will take some time to provide a simpler reproducible.

In the pointed test, it seems that there is no package name migration for one of the value inside the annotation.

The use case, i spotted is inside a package-info.java file with this content:
```
@XmlSchema(namespace = "https://www.lfenergy.org/compas/example", elementFormDefault = javax.xml.bind.annotation.XmlNsForm.QUALIFIED)
package org.lfenergy.compas.core.commons.model;

import javax.xml.bind.annotation.XmlSchema;
```

that the migration resulted in:
```
@XmlSchema(namespace = "https://www.lfenergy.org/compas/example", elementFormDefault = javax.xml.bind.annotation.XmlNsForm.QUALIFIED)
package org.lfenergy.compas.core.commons.model;

import jakarta.xml.bind.annotation.XmlSchema;
```

although I think it will be better if it is rewriting like:
```
@XmlSchema(namespace = "https://www.lfenergy.org/compas/example", elementFormDefault = jakarta.xml.bind.annotation.XmlNsForm.QUALIFIED)
package org.lfenergy.compas.core.commons.model;

import jakarta.xml.bind.annotation.XmlSchema;
```

If I try to extract the information, i would say that it is a problem with this kind of case:
```
@AnAnnotation(key = my.package.to.rename.MyClass)
```

**Comment by timtebeek** [comment]
That's very helpful, thanks! Sorry about not fully understanding before, this seems like a very clear bug!

**Comment by timtebeek** [comment]
I've added tests to reproduce this issue in https://github.com/openrewrite/rewrite/pull/3204

It seems we skip the visitor execution for this recipe for FQCN only, likely due to the applicability tests. Applicability tests are getting an overhaul in this open branch https://github.com/openrewrite/rewrite/pull/3118, so I propose we take another look at fixing this issue as soon as that pull request has been merged.

In addition to not rewriting FQCNs we might also have a type reference issue with non-FQCN annotation arguments. Thanks for bringing both to our attention! :)

**Comment by timtebeek** [comment]
So 8.0 has been released, but so far the tests related to this issue have not been fixed because of that:
https://github.com/openrewrite/rewrite/blob/main/rewrite-java-test/src/test/java/org/openrewrite/java/ChangePackageTest.java#L559
We'll have to dive in now that we have a better grasp of the problem and 8.0-compatible solution.

**Comment by timtebeek** [comment]
So turns out there were some issues with the tests that aimed to replicate this issue; the tests pass with the fixes applied in #3294 . That hints at this issue being resolved when the required types can actually be resolved. Can you confirm that @apupier ? Or are you able to provide a sample where it's not yet working as expected?

**Comment by pativa** [comment]
This seems to still not work for us, so I have provided a test case that does not work in this PR: https://github.com/openrewrite/rewrite/pull/4174
